## 군집분석 1

In [2]:
m1 <- matrix(c(150,50,130,55,80,80,100,85,95,91),
            nrow = 5,
            ncol = 2,
            byrow = TRUE)

In [3]:
m1

[,1] [,2]
[1,] 150  50  
[2,] 130  55  
[3,]  80  80  
[4,] 100  85  
[5,]  95  91

In [5]:
D1 <- dist(m1)
D1

         1        2        3        4
2 20.61553                           
3 76.15773 55.90170                  
4 61.03278 42.42641 20.61553         
5 68.60029 50.20956 18.60108  7.81025

In [6]:
?dist

### dist는 데이터의 거리를 계산하는 함수이다. method의 default값은 유클리디안 거리이다. 

In [8]:
D2 <- dist(m1, method = 'minkowski', p=3)
D2

          1         2         3         4
2 20.103629                              
3 71.790544 52.002096                    
4 55.164795 37.797631 20.103629          
5 61.735957 44.736068 16.757812  6.986368

In [10]:
m2 <- matrix(c(20,6,14,30,7,15,46,4,2),
            nrow = 3,
            ncol = 3,
            byrow = TRUE,
            dimnames = list(c('obs1', 'obs2', 'obs3'),
                           c('age', 'exp', 'time')))
m2

age exp time
obs1 20  6   14  
obs2 30  7   15  
obs3 46  4    2

In [11]:
cor(m2[1,], m2[2,])

[1] 0.9673518

In [12]:
cor(m2[1,], m2[3,])

[1] 0.7984081

### 객체1(obs1)과 객체2의 유사성이 객체1과 객체3간의 유사성보다 크다 (0.9674 > 0.7984)